src/modules/tools/zprobe/CalibrationStrategy.h

```diff
@@ -0,0 +1,64 @@
+// Generic calibration strategy based on nonlinear optimization
+//
+// Copyright (c) Oskar Linde 2015
+//
+// Licence: GPL v2
+
+#ifndef _GENERICLEVELINGSTRATEGY
+#define _GENERICLEVELINGSTRATEGY
+
+#include "LevelingStrategy.h"
+#include "Vector3.h"
+#include <string>
+#include <vector>
+
+#define calibration_strategy_checksum CHECKSUM("calibration")
+
+class CalibrationStrategy : public LevelingStrategy
+{
+public:
+    CalibrationStrategy(class ZProbe *zprobe) : LevelingStrategy(zprobe){}
+    ~CalibrationStrategy(){}
+    bool handleGcode(Gcode* gcode);
+    bool handleConfig();
+
+    // probes n points, return actuator positions of trigger points
+    bool probe_pattern(int n, int repeats, float actuator_positions[/*N*/][3]);
```

Comment:
Not really. Just thought it could be useful from the outside if one wants do visualization of the bed results or similar in another module.